lib/llvm-backend/src/platform/unix.rs

```diff
@@ -68,3 +74,60 @@ extern "C" fn signal_trap_handler(
         throw_trap(2);
     }
 }
+
+pub unsafe fn alloc_memory(
+    size: usize,
+    protect: MemProtect,
+    ptr_out: &mut *mut u8,
+    size_out: &mut usize,
+) -> LLVMResult {
+    let size = round_up_to_page_size(size);
+    let ptr = mmap(
+        ptr::null_mut(),
+        size,
+        match protect {
+            MemProtect::NONE => PROT_NONE,
+            MemProtect::READ => PROT_READ,
+            MemProtect::READ_WRITE => PROT_READ | PROT_WRITE,
+            MemProtect::READ_EXECUTE => PROT_READ | PROT_EXEC,
+        },
+        MAP_PRIVATE | MAP_ANON,
+        -1,
+        0,
+    );
+    if ptr as isize == -1 {
+        return LLVMResult::ALLOCATE_FAILURE;
+    }
+    *ptr_out = ptr as _;
+    *size_out = size;
+    LLVMResult::OK
+}
+
+pub unsafe fn protect_memory(ptr: *mut u8, size: usize, protect: MemProtect) -> LLVMResult {
+    let res = mprotect(
+        ptr as _,
+        round_up_to_page_size(size),
+        match protect {
+            MemProtect::NONE => PROT_NONE,
+            MemProtect::READ => PROT_READ,
+            MemProtect::READ_WRITE => PROT_READ | PROT_WRITE,
+            MemProtect::READ_EXECUTE => PROT_READ | PROT_EXEC,
+        },
+    );
+
+    if res == 0 {
+        LLVMResult::OK
+    } else {
+        LLVMResult::PROTECT_FAILURE
+    }
+}
+
+pub unsafe fn dealloc_memory(ptr: *mut u8, size: usize) -> LLVMResult {
+    let res = munmap(ptr as _, round_up_to_page_size(size));
```

Comment:
You shouldn't need round_up_to_page_size here either (though it helps to keep the sizes to mmap and munmap the same).

Reply:
This was same code as before, just moved into a new file.
Perhaps we can address this in a separate PR?